Dasd 7575 Mac 65 migrate to app service

NickGraham101 opened the pull request.

NickGraham101 committed:
Converted CloudService to AppService

NickGraham101 committed:
Add ARM template for app service & other resources

NickGraham101 committed:
Added pipeline to build and deploy app service

NickGraham101 committed:
Adds AppInsights to worker

NickGraham101 committed:
Added AcceptanceTest job to pipeline

NickGraham101 committed:
Removed staging slot deployment for worker

EwanNoble commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/tags/0.4.55
+    endpoint: GitHub (SFA)
```
```suggestion
    endpoint: SkillsFundingAgency
```

NickGraham101 replied:
Updated in single commit

EwanNoble commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+    type: github
+    name: SkillsFundingAgency/das-platform-automation
+    ref: refs/tags/4.6.0
+    endpoint: GitHub (SFA)
```
```suggestion
    endpoint: SkillsFundingAgency
```

NickGraham101 replied:
Updated in single commit

EwanNoble commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+  - repository: das-platform-building-blocks
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/tags/0.4.55
```
```suggestion
    ref: refs/tags/0.4.57
```

NickGraham101 replied:
Updated in single commit

EwanNoble commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+  - repository: das-platform-automation
+    type: github
+    name: SkillsFundingAgency/das-platform-automation
+    ref: refs/tags/4.6.0
```
```suggestion
    ref: refs/tags/5.0.2
```

NickGraham101 replied:
Updated in single commit

adam230594 commented on pipeline-templates/job/deploy.yml:
```diff
@@ -0,0 +1,80 @@
+            ServiceConnection: ${{ parameters.ServiceConnection }}
+            AppServiceName: $(ApiAppServiceName)
+            DeploymentPackagePath: $(Pipeline.Workspace)/${{ parameters.SolutionBaseName }}/${{ parameters.SolutionBaseName }}.API/${{ parameters.SolutionBaseName }}.API.zip
+        - task: AzureRmWebAppDeployment@4
```
Is ZipDeploy deployment type not working for this app?

NickGraham101 replied:
Didn't work for the function app and is now broken for the web app as well, will try to identify why but might need to use and earlier version of the app-deploy template

NickGraham101 replied:
It doesn't build locally using dotnet core and looking back at the history in this [branch](https://github.com/SkillsFundingAgency/das-tasks/tree/DASD-7575-MigrateToAppServiceWebJob) (before I rebased and tidied up) it looks like I had a lot of problems getting that to work on the agent as well.  Publishing with `VSBuild@1` was the solution but this will mean using `webDeploy` instead of `zipDeploy`

adam230594 replied:
There is a solution for this by changing the arguments of the VsBuild Task. An example commit is here
 eg: https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/commit/90f6e6ad7a756c09a5fb43ff87cc4a6bb233bae6

NickGraham101 replied:
Build uses zipDeploy

adam230594 commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+  - repository: das-platform-building-blocks
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/tags/0.4.55
```
Later tag version available

NickGraham101 replied:
Updated in single commit

adam230594 commented on azure-pipelines.yml:
```diff
@@ -0,0 +1,79 @@
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/tags/0.4.55
+    endpoint: GitHub (SFA)
```
SkillsFundingAgency endpoint preferred

NickGraham101 replied:
Updated in single commit

NickGraham101 committed:
Updated pipeline resource service connections & bumped version

NickGraham101 committed:
Changed deployment method to zipDeploy

The app is using .NET Framework so needs to be built using MSBuild and
deployed using WedDeploy, for most similar apps this is supported using
a task group.  This is the only app that requires an Azure Pipelines
step template to run the deployment.

NickGraham101 committed:
Changed pipeline trigger to master only
Manage Accounts Ticket: MAC-65

adam230594 reviewed (dismissed):
ARM and YAML changes LGTM

EwanNoble commented on src/SFA.DAS.Tasks.API/SFA.DAS.Tasks.API.csproj:
```diff
@@ -179,8 +179,13 @@
+    </Reference>
     <Reference Include="System.Data" />
     <Reference Include="System.Data.DataSetExtensions" />
+    <Reference Include="System.Data.OracleClient" />
```
This looks like a strange addition 🤔

NickGraham101 replied:
That snuck in when I removed the cloud service in May last year, not sure why.  Removed reference.

EwanNoble commented on src/SFA.DAS.Tasks.Worker/ApplicationInsightsInitializer.cs:
```diff
@@ -0,0 +1,15 @@
+    {
+        public void Initialize(ITelemetry telemetry)
+        {
+            telemetry.Context.Properties["Application"] = "SFA.DAS.Support.Indexer.Worker";
```
```suggestion
            telemetry.Context.Properties["Application"] = "SFA.DAS.Tasks.Worker";
```

NickGraham101 replied:
Updated

NickGraham101 committed:
Removed unneeded package reference for System.Data.OracleClient

NickGraham101 committed:
Fixed name of app in TelemetryInitializer

NickGraham101 committed:
Added pipeline stages for TEST, TEST2 & PP

NickGraham101 committed:
Made elasticPoolName ARM param optional

NickGraham101 committed:
All deploy stages depends on build stage
